app/workers/recover_lost_jobs_worker.rb

```diff
@@ -9,5 +9,17 @@ def perform
     Message.unprocessed.where("created_at <= ?", 1.hour.ago).pluck(:id).each do |message_id|
       ProcessMessageWorker.perform_async(message_id)
     end
+
+    DigestRunSubscriber.unprocessed.where("created_at <= ?", 1.hour.ago).pluck(:id).each do |digest_run_subscriber_id|
+      DigestEmailGenerationWorker.perform_async(digest_run_subscriber_id)
+    end
+
+    DigestRun.daily.unprocessed.where("created_at <= ?", 1.hour.ago).pluck(:date).each do |date|
```

Comment:
I'm wondering whether an hour is enough time to allow the digest runs to process now we're moving towards a default model of digest emails as the burden on these workers will only increase.

I ran a query over the last 20 digest runs and the average processing time was over 31 minutes with a maximum of 52 minutes and minimum of 1 minute.

Maybe increase the threshold of the daily and weekly processing time to 2 hours?

Reply:
Yeah, I did wonder about that. I checked in production and the timings are similar to what you're suggesting here. My conclusion is that they're similar enough to those for processing content changes that we can retry them similarly after 1 hour. The main reason we would want to increase the grace period would be if it would otherwise cause a large / alarming build up on the queue, which won't happen for digest runs.

Reply:
> The main reason we would want to increase the grace period would be if it would otherwise cause a large / alarming build up on the queue, which won't happen for digest runs.

True! It just seems this code may be run unnecessarily in the not so distant future as processing times increase, not that it particularly matters as there will be no adverse side effects, as you pointed out.

Reply:
The somewhat benign adverse effect is that our stats could show multiple successful digest runs for a day if this kick off happens before the digest finishes when we hit our first 1 hour one (which is surely looming with the next experiment?)

A way we could resolve that while keeping this at 1 hour is change the uniqueness args in the worker to also use current date e.g: `[args.first || Date.current.to_s]` - bit hacky mind.

Reply:
That's a good thought. I don't think it'll be a problem, though: our conflict strategy is to `log`, so the duplicate job would never actually make it on to the queue i.e. would never be reported as a metric.

Reply:
It wouldn't recognise it as a conflict as I understand as the one that starts automatically would have arguments of [nil] whereas the one that retried would have arguments of `["2020-09-18"]`

Reply:
Oh, I see what you mean now. Yes, that's another surprise we may as well avoid. It makes sense to me, seeing the same defaults next to each other in the worker code. I've updated the second commit to do that - thanks!